pkg/provider/handlers/read.go

```diff
@@ -49,9 +50,15 @@ func MakeReadHandler(client *containerd.Client) func(w http.ResponseWriter, r *h
 				Secrets:     fn.secrets,
 				EnvVars:     fn.envVars,
 				EnvProcess:  fn.envProcess,
-				Limits:      &types.FunctionResources{Memory: memory.String()},
 				CreatedAt:   fn.createdAt,
-			})
+			}
+
+			limit := &types.FunctionResources{Memory: memory.String()}
+			if limit.Memory != "0" {
```

Comment:
Please add a comment (note) about why this is being added.

I could easily imagine me or someone else deleting this thinking it's unnecessary in the future.

Reply:
Done